File: wheystore/wheystore.Web/Modules/SanPhamDB/GioHang/GioHangColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.GioHang")]
[BasedOnRow(typeof(GioHangRow), CheckNames = true)]
public class GioHangColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string Username { get; set; }
    public string MaSpTenSp { get; set; }
    public int SoLuong { get; set; }
    public decimal GiaLucThem { get; set; }
    public DateTime NgayTao { get; set; }
    [EditLink]
    public string GhiChu { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/GioHangColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { GioHangRow } from "./GioHangRow";

export interface GioHangColumns {
    Id: Column<GioHangRow>;
    Username: Column<GioHangRow>;
    MaSpTenSp: Column<GioHangRow>;
    SoLuong: Column<GioHangRow>;
    GiaLucThem: Column<GioHangRow>;
    NgayTao: Column<GioHangRow>;
    GhiChu: Column<GioHangRow>;
}

export class GioHangColumns extends ColumnsBase<GioHangRow> {
    static readonly columnsKey = 'SanPhamDB.GioHang';
    static readonly Fields = fieldsProxy<GioHangColumns>();
}